est.nf

```diff
@@ -4,23 +4,37 @@ process get_sequence_ids {
         path 'import_stats.json', emit: 'import_stats'
         path 'sequence_metadata.tab', emit: 'sequence_metadata'
         path 'sunburst_ids.tab', emit: 'sunburst_ids'
+        path 'blast_hits.tab', emit: 'blast_hits'
     stub:
     """
     cp $existing_fasta_file allsequences.fa
     """
     script:
-    common_args = "--efi-config ${params.efi_config} --efi-db ${params.efi_db} --mode ${params.import_mode}"
-    if (params.import_mode == "family")
+    common_args = "--efi-config ${params.efi_config} --efi-db ${params.efi_db} --mode ${params.import_mode} --sequence-version ${params.family_id_format}"
+    if (params.import_mode == "family") {
         """
-        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --family ${params.families} --sequence-version ${params.family_id_format}
+        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --family ${params.families}
         """
-    else if (params.import_mode == "accessions") {
+    } else if (params.import_mode == "accessions") {
         """
-        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --accessions ${params.accessions_file}
+        perl $projectDir/src/est/import/get_sequence_ids.pl $common_args --user-accession-file ${params.accessions_file}
         """
-    }
-    else
+    } else if (params.import_mode == "blast") {
+        // blast_hits.tab is provided as an output to the user
+        """
+        blastall -p blastp -i ${params.blast_query_file} -d ${params.fasta_db} -m 8 -e ${params.blast_evalue} -b ${params.num_blast_matches} -o init.blast
+        SZ=\$(wc -c "init.blast" | cut -d' ' -f1)
+        if [[ \$SZ != 0 ]]; then
+            cat init.blast | grep -v '#' | cut -f 1,2,3,4,12 | sort -k5,5nr > init_blast.tab
+            cat init.blast | grep -v '#' | cut -f 2,11 | sort -k2nr > blast_hits.tab
```

Comment:
Yes, `cat` should be removed to eliminate the extra pipe.  The `grep` and `cut` could likely be replaced with a single `awk` statement as well.  I haven't thought about the necessity of the sort command.  I will look into it.  The memory point you raised is likely not a problem since the file will be around ~1000 lines long.

Reply:
That sounds perfect